### DISCUSSION TITLE
Right Click Context Menu

### DISCUSSION THREAD

**Discussion by tonykastaneda** [johnlindquist/kit#383] [Ideas]
Not sure if this is the right category but would it be possible to creat a script that would open up the right click context menu over a file in finder without having the mouse over said file?

**Comment by johnlindquist** [comment]
Couple things:

## `getSelectedFile`
I usually use "getSelectedFile" which returns the filePath of selected file in Finder:

```js
let filePath = await getSelectedFile()

console.log(filePath) // logs the filePath of the selected file
await wait(2000)
```

This lets me do whatever I want with the `filePath` variable in the rest of the script. read/write/upload/etc

## `applescript`

If you need the full context menu, it requires applescript. Here's an example to trigger the "Show in enclosing folder" action from the context menu:

```js
await applescript(`
tell application "System Events" to tell application process "Finder"
    set selectedFile to value of attribute "AXFocusedUIElement"
    tell selectedFile to perform action "AXShowMenu"
    delay 0.5
    keystroke "Show in enclosing"
    -- Sends enter
    key code 36
end tell
`)
```

Does that sufficiently answer your question?

**Reply by tonykastaneda** [reply]
Hmm probably not what I’m looking for bit I’m also not that well versed in AppleScript. I want to be able to open up finder use the arrow keys to highlight a file/folder. I then a simple key stroke that will trigger the right click prompt as it would if I had moused over and right clicked on the file/folder.

**Reply by johnlindquist** [reply]
This will allow you to open the right-click menu by pressing "opt+r". You can change the shortcut to whatever.

```js
// Shortcut: opt r

await applescript(`
tell application "System Events" to tell application process "Finder"
    set selectedFile to value of attribute "AXFocusedUIElement"
    tell selectedFile to perform action "AXShowMenu"
end tell
`)
```

**Reply by johnlindquist** [reply]
Maybe I was confusing when I said "applescript". I meant use the applescript helper function to run applescript. These are still scripts which you can run from Script Kit.

**Reply by tonykastaneda** [reply]
Thank you 🙏
